apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -39,6 +44,34 @@ export const getPageById = (id: number) => {
     .executeTakeFirstOrThrow()
 }
 
+export const updatePageById = (
+  page: Partial<Omit<Page, 'id' | 'blobId'>> & { id: string },
+) => {
+  const { id, ...rest } = page
+  return db.transaction().execute((tx) => {
+    return tx
+      .updateTable('Resource')
+      .set(rest)
+      .where('id', '=', id)
+      .executeTakeFirstOrThrow()
+  })
+}
+
+export const updateBlobById = (props: UpdateBlobProps) => {
+  const { id, content } = props
+  return db.transaction().execute((tx) => {
+    return (
+      tx
+        .updateTable('Blob')
+        .innerJoin('Resource', 'Resource.id', 'id')
+        // NOTE: This works because a page has a 1-1 relation with a blob
+        .set({ content })
```

Comment:
question(blocking): hmm this feels weird, looks like it is only saving a JSON blob with only the `content` key. Should we just directly store the object instead, and not wrap it inside a `content` key?

This means the DB stores the blob content like this:
```
{
  "version": "0.1.0",
  "layout": "article",
  ...
}
```

Instead of:
```
{
  "content": {
    "version": "0.1.0",
    "layout": "article",
    ...
  }
}
```

Reply:
need your take on this - i'm not quite sure what content implies, i'm taking reference from `placeholder` (see `Preview.tsx`) 

i did this initially because the `placeholder.json` has more properties beyond just `content` (iirc), so in the future if we want to save those properties can just add here

Reply:
Ah yeah I meant like we should be saving all the props exactly like how `placeholder.json` looks like inside the DB. Right now I'm not sure:

1. What is inside the `content` key (is it `IsomerPageSchema` or `IsomerPageSchema['content']`
2. Are we storing `layout`, `version`, etc. somewhere else or something

Reply:
ah ok, it's `IsomerPageSchema` - this is confusing because we've chosen `content` to represent our blobs (but at the same time, the page schema itself has another key called `content`)

Reply:
Can ignore, clarified offline. The `content` here refers to the `content` column inside the DB table.